src/traces/mesh3d/convert.js

```diff
@@ -52,7 +53,69 @@ proto.handlePick = function(selection) {
 };
 
 function parseColorArray(colors) {
-    return colors.map(str2RgbaArray);
+    var b = [];
+    var len = colors.length;
+    for(var i = 0; i < len; i++) {
+        b[i] = str2RgbaArray(colors[i]);
+    }
+    return b;
+}
+
+// Unpack position data
+function toDataCoords(axis, coord, scale, calendar) {
+    var b = [];
+    var len = coord.length;
+    for(var i = 0; i < len; i++) {
+        b[i] = axis.d2l(coord[i], 0, calendar) * scale;
+    }
+    return b;
+}
+
+// round indices if passed as floats
+function toIndex(a) {
+    var b = [];
+    var len = a.length;
+    for(var i = 0; i < len; i++) {
+        b[i] = Math.round(a[i]);
+    }
+    return b;
+}
+
+// create cells
+function delaunayCells(delaunayaxis, positions) {
+    var d = ['x', 'y', 'z'].indexOf(delaunayaxis);
+    var b = [];
+    var len = positions.length;
+    for(var i = 0; i < len; i++) {
+        b[i] = [positions[i][(d + 1) % 3], positions[i][(d + 2) % 3]];
+    }
+    return triangulate(b);
+}
+
+// validate indices
+function hasValidIndices(list, numVertices) {
+    var len = list.length;
+    for(var i = 0; i < len; i++) {
+        if(!isIndex(list[i], numVertices)) {
+            return false;
+        }
+    }
+    return true;
+}
+
+// avoid pointing to an identical vertex twice
+function isTriangle(lists) {
```

Comment:
This criterion looks new. Was this leading to any bugs before?

Reply:
No bugs before. So I simply removed that check.